data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -210,11 +214,79 @@ class ApiServerClient {
       });
   }
 
+  registerInterOperatorCrds() {
+    return Promise.all([
+      this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES),
+      this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS),
+      this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEINSTANCES),
+      this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS)
+    ]);
+  }
+
   getCrdJson(resourceGroup, resourceType) {
     const crdEncodedTemplate = config.apiserver.crds[`${resourceGroup}_${CONST.APISERVER.API_VERSION}_${resourceType}.yaml`];
     logger.debug(`Getting crd json for: ${resourceGroup}_${CONST.APISERVER.API_VERSION}_${resourceType}.yaml`);
     return yaml.safeLoad(Buffer.from(crdEncodedTemplate, 'base64'));
   }
+  /**
+   * @description Create Namespace of given name
+   * @param {string} name - Name of resource group ex. backup.servicefabrik.io
+   */
+  createNamespace(name) {
+    assert.ok(name, `Property 'name' is required to create namespace`);
+    if (name === CONST.APISERVER.DEFAULT_NAMESPACE) {
+      return;
+    }
+    const resourceBody = {
+      kind: 'Namespace',
+      apiVersion: CONST.APISERVER.NAMESPACE_API_VERSION,
+      metadata: {
+        name: name
+      }
+    };
+    return Promise.try(() => this.init())
+      .then(() => apiserver
+        .api[CONST.APISERVER.NAMESPACE_API_VERSION].ns.post({
+          body: resourceBody
+        }))
+      .tap(() => logger.debug(`Successfully created namespace ${name}`))
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      })
+      .catch(Conflict, () => logger.debug(`Namespace ${name} already exists, proceeding...`));
+  }
+
+  deleteNamespace(name) {
+    return Promise.try(() => this.init())
+      .then(() => apiserver
+        .api[CONST.APISERVER.NAMESPACE_API_VERSION].ns(name).delete())
+      .tap(() => logger.debug(`Successfully deleted namespace ${name}`))
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      });
+  }
+
+  getNamespaceId(resourceId) {
+    return _.get(config, 'apiserver.enable_resource_isolation') ? `sf-${resourceId}` : CONST.APISERVER.DEFAULT_NAMESPACE;
```

Comment:
For BOSh based deployments, we have not enabled because of following reason.

1. Ideally, there is no resource isolation needed as it is managed centrally.
2. The migration of existing resource need to be done, which will be a huge task in itself.

Reply:
Let's discuss on the name change. :)

Regarding your question on the segregation, on k8s landscape even the SF resources are separated in different namespaces.

Reply:
`enable_name_space` or `use_default_name_space`

Reply:
Changed it to `enable_namespace `